vkcs/mlplatform/resource_spark_k8s.go

```diff
@@ -0,0 +1,502 @@
+		Keypair:              data.Keypair.ValueString(),
+		DeleteAfterDelay:     int(data.DeleteAfterInactiveMin.ValueInt64()) * 60,
+		SuspendAfterDelay:    int(data.SuspendAfterInactiveMin.ValueInt64()) * 60,
+		SparkConfiguration:   sparkConfigurationEncoded,
+		EnvironmentVariables: environmentVariablesEncoded,
+	}
+
+	tflog.Debug(ctx, "Creating Spark K8S Cluster", map[string]interface{}{"options": fmt.Sprintf("%#v", sparkK8SCreateOpts)})
+
+	sparkK8SResp, err := clusters.Create(mlPlatformClient, &sparkK8SCreateOpts).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_mlplatform_spark_k8s", err.Error())
+		return
+	}
+	resp.State.SetAttribute(ctx, path.Root("id"), sparkK8SResp.ID)
+
+	sparkK8SStateConf := &retry.StateChangeConf{
+		Pending:    []string{clusterStatusCreating, clusterStatusInstallSparkOperator, clusterStatusProvisioning, clusterStatusStarting, clusterStatusCreatingRegistry},
+		Target:     []string{clusterStatusRunning},
+		Refresh:    clusterStateRefreshFunc(mlPlatformClient, sparkK8SResp.ID),
+		Timeout:    timeout,
+		Delay:      clusterDelay,
+		MinTimeout: clusterMinTimeout,
+	}
+
+	tflog.Debug(ctx, "Waiting for the Spark K8S cluster to be created", map[string]interface{}{"timeout": timeout})
+
+	_, err = sparkK8SStateConf.WaitForStateContext(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("Error waiting for the cluster to become ready", err.Error())
+		return
+	}
+
+	sparkK8SResp, err = clusters.Get(mlPlatformClient, sparkK8SResp.ID).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving vkcs_mlplatform_spark_k8s", err.Error())
+		return
+	}
+
+	data.ID = types.StringValue(sparkK8SResp.ID)
+	data.Region = types.StringValue(region)
+
+	data.Name = types.StringValue(sparkK8SResp.Name)
+	data.AvailabilityZone = types.StringValue(sparkK8SResp.Info.AvailabilityZone)
+	data.NetworkID = types.StringValue(sparkK8SResp.Info.NetworkID)
+	data.SubnetID = types.StringValue(sparkK8SResp.Info.SubnetID)
+	data.ClusterMode = types.StringValue(sparkK8SResp.Info.ClusterMode)
+	data.RegistryID = types.StringValue(sparkK8SResp.DockerRegistryID)
+	data.IPPool = types.StringValue(sparkK8SResp.Info.IPPool)
+	data.Keypair = types.StringValue(sparkK8SResp.Info.Keypair)
+
+	data.S3BucketName = types.StringValue(sparkK8SResp.S3BucketName)
+	data.HistoryServerURL = types.StringValue(sparkK8SResp.HistoryServerURL)
+	data.ControlInstanceID = types.StringValue(sparkK8SResp.ControlInstanceID)
+	data.InactiveMin = types.Int64Value(int64(sparkK8SResp.InactiveMin))
+	data.SuspendAfterInactiveMin = types.Int64Value(int64(sparkK8SResp.SuspendAfterInactiveMin))
+	data.DeleteAfterInactiveMin = types.Int64Value(int64(sparkK8SResp.DeleteAfterInactiveMin))
+
+	data.NodeGroups = flattenNodeGroupOpts(sparkK8SResp.Info.NodeGroups)
+
+	sparkConfiguration = sparkK8SResp.Info.SparkConfiguration
+	sparkConfigurationDecoded, _ := base64.StdEncoding.DecodeString(sparkConfiguration)
```

Comment:
Same thing as above, it would be better to place this logic into the client's Extract.

Reply:
Moved logic